stl/inc/format

```diff
@@ -1492,32 +1435,27 @@ class _Format_arg_store<_Context> {};
 
 template <class _Context>
 class basic_format_args {
-private:
-    template <class _Ty>
-    _NODISCARD static auto _Get_value_from_memory(const unsigned char* _Val) noexcept {
-        auto& _Temp = *reinterpret_cast<const unsigned char(*)[sizeof(_Ty)]>(_Val);
-        return _Bit_cast<_Ty>(_Temp);
-    }
-
 public:
     basic_format_args() noexcept;
     basic_format_args(const _Format_arg_store<_Context>&) noexcept {}
     template <class... _Args>
     basic_format_args(const _Format_arg_store<_Context, _Args...>& _Store) noexcept
-        : _Num_args(sizeof...(_Args)), _Storage(_Store._Storage) {}
+        : _Num_args(sizeof...(_Args)), _Index_array(_Store._Index_array) {}
 
     _NODISCARD basic_format_arg<_Context> get(const size_t _Index) const noexcept {
         if (_Index >= _Num_args) {
             return basic_format_arg<_Context>{};
         }
 
-        using _CharType                   = typename _Context::char_type;
-        const auto _Packed_index          = reinterpret_cast<const _Format_arg_store_packed_index*>(_Storage)[_Index];
-        const auto _Index_length          = _Num_args * sizeof(_Format_arg_store_packed_index);
-        const unsigned char* _Arg_storage = _Storage + _Index_length + _Packed_index._Index;
+        using _CharType          = typename _Context::char_type;
+        const auto _Packed_index = _Index_array[_Index];
+        const auto _Arg_storage =

```

Comment:
it would be good to note that we're reading off the end of the array here onto the next field in the object. (is this UB?)

Reply:
Pointer arithmetic to find the end of the array is well-defined, after that we access characters within the object representation of a trivially-copyable class. I think this all has defined behavior, if not quite portable since some implementations could put padding between the index array and the arg storage.